plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterManagerImpl.java

```diff
@@ -434,7 +434,8 @@ private IpAddress getSourceNatIp(Network network) {
     }
 
     public VMTemplateVO getKubernetesServiceTemplate(DataCenter dataCenter, Hypervisor.HypervisorType hypervisorType) {
-        VMTemplateVO template = templateDao.findSystemVMReadyTemplate(dataCenter.getId(), hypervisorType);
+        ConfigKey<String> preferredArchitecture = ResourceManager.SystemVmPreferredArchitecture;
+        VMTemplateVO template = templateDao.findSystemVMReadyTemplate(dataCenter.getId(), hypervisorType, preferredArchitecture.value());
```

Comment:
@nvazquez we need to use valueIn() method with zoneid for the config value.

Reply:
Fixed, thanks @harikrishna-patnala